src/viking.js

```diff
@@ -1,11 +1,103 @@
 // Soldier
-function Soldier() {}
+function Soldier(health, strength) {
+  this.health = health;
+  this.strength = strength;
+}
+
+Soldier.prototype.attack = function() {
+  return this.strength;
+}
+
+Soldier.prototype.receiveDamage = function(damage) {
+  this.health -= damage;
+}
 
 // Viking
-function Viking() {}
+function Viking(name, health, strength) {
+  this.name = name; 
+  Soldier.call(this, health, strength);
+}
+
+Viking.prototype = Object.create(Soldier.prototype);
+Viking.prototype.constructor = Viking;
+
+Viking.prototype.receiveDamage = function(damage) {
+  this.health -= damage;
+  if (this.health > 0) {
+    return `${this.name} has received ${damage} points of damage`;
+  } else {
+    return `${this.name} has died in act of combat`;
+  }
+}
+
+ Viking.prototype.battleCry = function() {
+   return "Odin Owns You All!";
+ }
 
 // Saxon
-function Saxon() {}
+function Saxon(health, strength) {
+  Soldier.call(this, health, strength);
+}
+
+Saxon.prototype = Object.create(Soldier.prototype);
+Saxon.prototype.constructor = Saxon;
+
+Saxon.prototype.receiveDamage = function(damage) {
+  this.health -= damage;
+  if (this.health > 0) {
+    return `A Saxon has received ${damage} points of damage`;
+  } else {
+    return "A Saxon has died in combat";
+  }
+}
 
 // War
-function War() {}
+function War() {
+ this.vikingArmy = [];
+ this.saxonArmy = [];
+
+ this.addViking = function(Viking){
+   this.vikingArmy.push(Viking);
+ };
+
+ this.addSaxon = function(Saxon){
+   this.saxonArmy.push(Saxon);
+ };
+
+ this.vikingAttack = function(){
+  let randomSaxonIndex = Math.floor(Math.random() * this.saxonArmy.length);
```

Comment:
This lines of code in this method, and the ones inside the next method are very similar so maybe we could encapsulate that into a function and reused :)